### DISCUSSION TITLE
How to put text in middle of donut

### DISCUSSION THREAD

**Discussion by jamiewhitham** [RomRider/apexcharts-card#248] [Q&A] [answered]
Hello

I would like to put some text in the middle of a donut chart. I tried looking at options->plotOptions->pie->donut->labels but couldn't get anything displayed despite no error.

Anyone have another way of doing this?
<img width="242" alt="total here" src="https://user-images.githubusercontent.com/6522767/144099042-e4977e76-82f0-4085-914e-7eb1589111ba.png">

**Comment by paulius2k** [comment, marked as answer]
Put on the _first_ indent level:
```
type: custom:apexcharts-card
chart_type: donut
apex_config:
  plotOptions:
    pie:
      donut:
        labels:
          show: true,
          total:
            showAlways: true,
            show: true
```

you will get this:
<img src="https://user-images.githubusercontent.com/24705262/144910830-a2a75b50-6fd4-40e2-b096-494788a1a47a.png" width=40%>

**Reply by mdoliver** [reply]
Hi @paulius2k 
Would you happen to know how to use the formatter function to prevent the extraneous decimals from displaying in the centre total (see screen shot)? I have the below code for the data labels inside the donut but not sure what to use for the centre total. Thanks!

`formatter: EVAL:function(val) {return (val).toFixed(1) + " %"}`

![image](https://user-images.githubusercontent.com/51429339/150875607-2b806baf-3a8e-4b7c-96b0-ec748144568a.png)

**Reply by paulius2k** [reply]
Hi @mdoliver 
I find this info from various sources:
1. https://apexcharts.com/docs/options/plotoptions/pie/#labels

>formatter: function
A custom formatter function to apply on the total value. It accepts one parameter w which contains the chart’s config and global objects.
Defaults to a total of all series percentage divided by the length of series.

and

2. https://github.com/apexcharts/apexcharts.js/issues/1262#issue-556469321

so just re-using it:
```
type: custom:apexcharts-card
chart_type: donut
apex_config:
  plotOptions:
    pie:
      donut:
        labels:
          show: true,
          total:
            showAlways: true,
            show: true
            formatter: |
              EVAL:function(w) {
                return w.globals.seriesTotals.reduce((a, b) => {return a + b}, 0).toLocaleString();
              }

```

seems to work for me.

Without `formatter:`:
<img src="https://user-images.githubusercontent.com/24705262/150953368-ce967afd-e094-47bf-be9e-e52df66a6e44.png" width=40%>


**with** the above `formatter:`:
<img src="https://user-images.githubusercontent.com/24705262/150953471-2c760258-ce1f-4d05-9f5a-75b86f7a2c34.png" width=40%>


I hope this works for you as well.

**Reply by mdoliver** [reply]
Thanks so much for your help! That's worked well for me.
